channels/ON-CHAIN.md

```diff
@@ -0,0 +1,372 @@
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| initiator_amount     | 32 |
+ ---------------------- ----
+| responder_amount     | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`: channel id as recorded on chain
+- `initiator_amount`: final balance for the initiator
+- `responder_amount`: final balance for the responder
+- `ttl`:
+- `fee`:
+- `nonce`: taken from the `initiator` account
+
+`initiator_amount` and `responder_amount` are the agreed upon distribution of
+coins. To get the final outcome of the channel, the fees have to get accounted
+for. It is advised that peers choose an even amount, otherwise one peer will end
+up paying more, although this should not be an issue in practice.
+
+```
+if initiator_amount + responder_amount < fee
+  return error
+else if initiator_amount >= ceil(fee/2) && responder_amount >= floor(fee/2)
+  initiator_final := initiator_amount - ceil(fee/2)
+  responder_final := responder_amount - floor(fee/2)
+else if responder_amount >= ceil(fee/2) && initiator_amount >= floor(fee/2)
+  responder_final := responder_amount - ceil(fee/2)
+  initiator_final := initiator_amount - floor(fee/2)
+else if initiator_amount > responder_amount
+  initiator_final := initiator_amount - fee + responder_amount
+  responder_final := 0.0
+else
+  responder_final := responder_amount - fee + initiator_amount
+  initiator_final := 0.0
+```
+
+
+
+#### Requirements
+
+This transaction MUST have valid signatures of all involved parties.
+
+After this transaction has been included in a block, the channel MUST be
+considered closed and allow no further modifications.
+
+
+### `channel_close_solo`
+
+In order to close a channel unilaterally, a user has to send a
+`channel_close_solo` transaction. This is only necessary if one peer stops
+responding but can also be used by an malicious peer trying to close a channel
+with a state that hasn't been agreed on by all participants.
+
+With the inclusion of this transaction on chain, the timer, during which
+disputes in the form of `channel_slash` will be considered, is started.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| from                 | 32 |
+ ---------------------- ----
+| amount               | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`:
+- `from`
+- `ttl`
+- `nonce`
+- `fee`
+
+
+### `channel_slash`
+
+If a malicious party sent a `channel_close_solo` with an outdated state, the
+honest party has the opportunity to issue a `channel_slash` transaction. This
```

Comment:
Can anyone? Or only a participant in the target channel distinct from the one that submitted `channel_close_solo` ?

Reply:
Both `slash` and `close_solo` transactions must be sent by channel parties

Reply:
The [clarification in channel_close_solo](https://github.com/aeternity/protocol/pull/72/commits/1eb7f5ec183dc75b56c809379b2adcead45e54e7#diff-4fbc1ee2b0f1c1c88eeeafc062a7989cR235) clarified this too. So this is addressed. Thanks.